src/utils/orgUtils.ts

```diff
@@ -65,6 +83,97 @@ export class OrgUtils {
         }
     }
 
+    public static async getCompactLayoutsForSObject(
+        sObjectName: string
+    ): Promise<SObjectCompactLayouts> {
+        const org = await Org.create();
+        const conn = org.getConnection();
+
+        const result = await conn.request<SObjectCompactLayouts>(
+            `/services/data/v59.0/sobjects/${sObjectName}/describe/compactLayouts`
+        );
+
+        return Promise.resolve(result);
+    }
+
+    public static async getCompactLayoutForSObject(
+        sObjectName: string,
+        recordTypeId: string
+    ): Promise<Object> {
+        const org = await Org.create();
+        const conn = org.getConnection();
+
+        const result = await conn.request<Object>(
+            `/services/data/v59.0/sobjects/${sObjectName}/describe/compactLayouts/${recordTypeId}`
+        );
+
+        return Promise.resolve(result);
+    }
+
+    public static async getCompactLayoutFieldsForSObject(
+        sObjectName: string
+    ): Promise<CompactLayoutField[]> {
+        try {
+            const fields: CompactLayoutField[] = [];
+
+            // Get all the compact layouts associated to this sObject first
+            let result = await this.getCompactLayoutsForSObject(sObjectName);
+
+            if (result) {
+                // sObject can have multiple compact layouts associated with it. Get the default.
+                const defaultCompactLayoutId = result.defaultCompactLayoutId;
+                result['defaultCompactLayoutId'];
+
+                // Mapping tab
+                const recordTypeCompactLayoutMappings =
+                    result.recordTypeCompactLayoutMappings;
+
+                // ID of compact layout need to be normalized
+                const recordTypeCompactLayoutMapping =
+                    recordTypeCompactLayoutMappings.find((element) => {
+                        if (defaultCompactLayoutId) {
+                            return (
+                                element.compactLayoutId ===
+                                defaultCompactLayoutId
+                            );
+                        } else {
+                            // defaultCompactLayoutId can be null when a compact layout is not assigned.
+                            // In that case sObject will always have one default layout called SYSTEM.
+                            // So use that instead.
+                            const compactLayoutName = element[
+                                'compactLayoutName' as keyof Object
```

Comment:
Can we update all of these, based on the previous feedback? We should be able avoid coercing a key type and having to cast properties to `unknown`.

- `compactLayoutName` is already in the [`SObjectCompactLayoutMapping` type](https://github.com/salesforce/salesforcedx-vscode-mobile/pull/53/files#diff-9059df4604057fed56162133c1646eececc243a7f261ac69ffb90d57918c0f70R28-R31).
- `recordTypeId`, used [below](https://github.com/salesforce/salesforcedx-vscode-mobile/pull/53/files#diff-9059df4604057fed56162133c1646eececc243a7f261ac69ffb90d57918c0f70R151-R154), is too.
- My read of what comes back from `getCompactLayoutForSObject()` is:
    - An `SObjectCompactLayoutMapping` instance, and then that type requires:
    - An additional `fieldItems` property of type `CompactLayoutField[]`, based on [the code below](https://github.com/salesforce/salesforcedx-vscode-mobile/pull/53/files#diff-9059df4604057fed56162133c1646eececc243a7f261ac69ffb90d57918c0f70R164-R166).  
    
    We should update `getCompactLayoutForSObject()` to return `Promise<SObjectCompactLayoutMapping>` accordingly, and that should close the loop.

Reply:
Updated!